test/proposal-genericscheme.spec.ts

```diff
@@ -33,30 +32,36 @@ describe('Proposal', () => {
     const states: IProposalState[] = []
     const lastState = (): IProposalState => states[states.length - 1]
 
-    const actionMockABI = arc.getABI(undefined, 'ActionMock', LATEST_ARC_VERSION)
+    const actionMockABI = arc.getABI({abiName: 'ActionMock', version: LATEST_ARC_VERSION})
 
     if(!arc.web3) throw new Error('Web3 provider not set')
 
-    const actionMock = new Contract(testAddresses.test.ActionMock.toString(), actionMockABI, arc.web3.getSigner())
     const callData = new ethers.utils.Interface(actionMockABI).functions.test2.encode([dao.id])
 
-    const schemes = await dao.schemes({ where: {name: 'GenericScheme' }}).pipe(first()).toPromise()
-    const genericScheme = schemes[0].coreState as ISchemeState
-    const proposal = await createAProposal(dao, {
+    const plugins = await dao.plugins({ where: {name: 'GenericScheme' }}).pipe(first()).toPromise() as GenericScheme[]
+    const genericScheme = plugins[0]
+
+    const tx = await genericScheme.createProposal({
+      dao: dao.id,
       callData,
-      scheme: genericScheme.address,
-      schemeToRegister: actionMock.address,
-      value: 0
-    })
+      value: 0,
+      plugin: getTestScheme('GenericScheme')
```

Comment:
@namesty why do we need to pass the "plugin" argument here in `createProposal`? Is this not just the genericScheme's address, i.e. available in the state of `genericScheme`?

Reply:
Agreed. It uses the plugin ID which is available in the plugin instance. I still kept the plugin parameter as optional, if not passed it will take the instance's id.

Reply:
:+1: